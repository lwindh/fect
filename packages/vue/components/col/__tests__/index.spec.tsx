import { mount } from '@vue/test-utils'
import Col from '../index'
import Row from '../../row'

describe('Col', () => {
  it('should be render as Element', () => {
    const col = mount(Col)
    expect(col.html()).toMatchSnapshot()
    expect(() => col.unmount()).not.toThrow()
  })

  it('should be support tag', () => {
    const wrapper = mount(Col, {
      props: {
        tag: 'span',
      },
    })
    expect(wrapper.find('span')).toBeTruthy()
  })
  it('should be use row gutter', () => {
    const wrapper = mount({
      render() {
        return (
          <Row gutter={1}>
            <Col />
          </Row>
        )
      },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
