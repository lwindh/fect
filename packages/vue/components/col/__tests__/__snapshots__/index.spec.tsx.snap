// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Col should be render as Element 1`] = `
<div style="width: 100%; margin-left: 0%; padding-left: 0px; padding-right: 0px;" class="fect-col">
  <!---->
</div>
`;

exports[`Col should be use row gutter 1`] = `
<div class="fect-row " style="margin-left: -0.5px; margin-right: -0.5px;">
  <div style="width: 100%; margin-left: 0%; padding-left: 0.5px; padding-right: 0.5px;" class="fect-col">
    <!---->
  </div>
</div>
`;
